tests/tokens/test_token_methods.py

```diff
@@ -14,6 +14,14 @@ class Test(IToken):
     return Test()
 
 
+def test_if_token_is_allowed():
+    test_token: IToken = create_token()
+    # allowed only if both SYMBOL_RATE and NETWORK_IDENTIFIER exist.
+    assert test_token.is_allowed({"T_RATE": 1000}, {"Test", "Another"})
+    assert not test_token.is_allowed({"A_RATE": 1000}, {"Test", "Another"})
+    assert not test_token.is_allowed({"T_RATE": 1000}, {"Another"})
```

Comment:
I do not yet get why this would return false as the first one is true and is basically the same with one extra network

Reply:
There is nothing that says "Another" has a symbol of "A".
This is why L21 is false. Because there is nothing corresponding to "A_RATE".
L22 is false because again "T_RATE" doesn't correspond to "Another"